core/src/main/java/com/opentext/ia/sdk/sip/BatchSipAssembler.java

```diff
@@ -85,7 +85,8 @@ public synchronized void add(D domainObject) throws IOException {
   }
 
   private boolean shouldStartNewSip(D component) {
-    return current == null || segmentationStrategy.shouldStartNewSip(component, assembler.getMetrics());
+    // We need to check that the DomainObject will fit in the Sip before we return anything

```

Comment:
1. This comment is too specific. At this point in the code you don't know what the segmentation strategy does, so don't mention that.
1. I guess you reversed the conditions to allow the segmentation strategy to throw an exception? If so, then mention that so that future maintainers aren't tempted to reverse it back.

Reply:
This is bad on my part. I reversed it because we don't need to start a new sip EVERY time the current == null, because the object may be too big for the sip anyway.

So I'm basing this on the fact I know other parts of the code. But every method should work in isolation ideally, so this is a bad thing to do in that respect. What do you think I should do?

Reply:
If `current == null`, does your segmentation strategy throw an exception when the contents for the domain object are too big? That could be a valid reason to keep it this way. With the original order, when `current == null` the segmentation strategy would never get a chance because of short circuit boolean evaluation.

Reply:
No it doesn't because it returns with the true immediately, starts the new SIP, and then adds the DomainObject without checking. This is because the old code ALWAYS adds the first SIP.
Because my code looks forward sometimes the first SIP won't fit. So we always need to check that it'll fit before starting the SIP.